### PR TITLE
Prepare for the removal of the securitymanager in moxy.

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Classes to be removed:
- PrivilegedClassForName
- PrivilegedMethodInvoker
- PrivilegedNewInstanceFromClass

### CODE REVIEW

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/internal/jaxb/XMLJavaTypeConverter.java [hunk 1]
```diff
@@ -51,12 +48,12 @@
 public class XMLJavaTypeConverter extends org.eclipse.persistence.oxm.mappings.converters.XMLConverterAdapter {
     protected Class<?> boundType = Object.class;
     protected Class<?> valueType = Object.class;
-    protected Class<?> xmlAdapterClass;
+    protected Class<XmlAdapter<?,?>> xmlAdapterClass;
```
Comment (lukasj) [inline moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/internal/jaxb/XMLJavaTypeConverter.java:51]: this should be `Class<? extends XmlAdapter<?,?>>` instead.

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/plugins/BeanValidationPlugin.java [hunk 2]
```diff
@@ -209,7 +206,7 @@ public int parseArgument(Options opt, String[] args, int i) throws BadCommandLin
 
     /* ######### CORE FUNCTIONALITY ######### */
     private static final String PATTERN_ANNOTATION_NOT_APPLICABLE = "Facet \"pattern\" was detected on a DOM node with non-string base type. Annotation was not generated, because it is not supported by the Bean Validation specification.";
-    private final boolean securityEnabled = System.getSecurityManager() != null;
+    //private final boolean securityEnabled = System.getSecurityManager() != null;
```
Comment (lukasj) [inline moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/plugins/BeanValidationPlugin.java:209]: remove?

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/internal/jaxb/XMLJavaTypeConverter.java (hunk 1 above)
Comment (Tomas-Kraus) [reply moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/internal/jaxb/XMLJavaTypeConverter.java:51]: Yes, makes sense. Will fix with next iteration.

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/plugins/BeanValidationPlugin.java (hunk 2 above)
Comment (Tomas-Kraus) [reply moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/plugins/BeanValidationPlugin.java:209]: Sure. :) Just keep in mind that it's 'in progress' quality changeset.

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java [hunk 3]
```diff
@@ -62,60 +60,44 @@ public Method getBeforeUnmarshalCallback() {
...
-                }else{
-                    afterUnmarshal = PrivilegedAccessHelper.getMethod(domainClass, "afterUnmarshal", params, false);
-                }
+                Method afterUnmarshal = PrivilegedAccessHelper.callDoPrivilegedWithException(
+                        () -> PrivilegedAccessHelper.getMethod(domainClass, "afterUnmarshal", params, false)
+                );
                 setAfterUnmarshalCallback(afterUnmarshal);
-            } catch (NoSuchMethodException nsmex) {}
+            } catch (NoSuchMethodException nsmex) {
+                AbstractSessionLog.getLog().logThrowable(SessionLog.FINE, nsmex);
```
Comment (lukasj) [inline moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java:97]: is this needed? `afterUnmarshal` is NOT mandatory and in most cases will be missing, so the log will become huge

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java [hunk 4]
```diff
@@ -62,60 +60,44 @@ public Method getBeforeUnmarshalCallback() {
...
-                }else{
-                    beforeUnmarshal = PrivilegedAccessHelper.getMethod(domainClass, "beforeUnmarshal", params, false);
-                }
+                Method beforeUnmarshal = PrivilegedAccessHelper.callDoPrivilegedWithException(
+                        () -> PrivilegedAccessHelper.getMethod(domainClass, "beforeUnmarshal", params, false)
+                );
                 setBeforeUnmarshalCallback(beforeUnmarshal);
-            } catch (NoSuchMethodException nsmex) {}
+            } catch (NoSuchMethodException nsmex) {
+                AbstractSessionLog.getLog().logThrowable(SessionLog.FINE, nsmex);
```
Comment (lukasj) [inline moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java:85]: same as `afterUnmarshal`

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java (hunk 3 above)
Comment (Tomas-Kraus) [reply moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java:97]: OK, no logging here. Done in MarshalCallback too.

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java (hunk 4 above)
Comment (Tomas-Kraus) [reply moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/compiler/UnmarshalCallback.java:85]: OK, no logging here. Done in MarshalCallback too.

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/plugins/BeanValidationPlugin.java (hunk 2 above)
Comment (Tomas-Kraus) [reply moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/plugins/BeanValidationPlugin.java:209]: Removed

File: moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/internal/jaxb/XMLJavaTypeConverter.java (hunk 1 above)
Comment (Tomas-Kraus) [reply moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/internal/jaxb/XMLJavaTypeConverter.java:51]: Removed.

**Review by lukasj — APPROVED** [review]
LGTM

**Pull request merged**